xrdp/xrdp_mm.c

```diff
@@ -1218,15 +1246,27 @@ xrdp_mm_egfx_caps_advertise(void *user, int caps_count,
                 best_pro_index = index;
                 break;
             case XR_RDPGFX_CAPVERSION_10:
+                if (!(flags & XR_RDPGFX_CAPS_FLAG_AVC_DISABLED))
+                {
+                    best_h264_index = index;
+                }
                 best_pro_index = index;
                 break;
             case XR_RDPGFX_CAPVERSION_101:
                 best_pro_index = index;
                 break;
             case XR_RDPGFX_CAPVERSION_102:
+                if (!(flags & XR_RDPGFX_CAPS_FLAG_AVC_DISABLED))
```

Comment:
The code for 102, 103, 105, 106, and 107 all look identical... Perhaps we just consolidate those into one "fall through" switch block?

Reply:
8 & 101 are the same as well. -